proto/tendermint/abci/types.proto

```diff
@@ -21,20 +21,24 @@ import "gogoproto/gogo.proto";
 
 message Request {
   oneof value {
-    RequestEcho               echo                 = 1;
-    RequestFlush              flush                = 2;
-    RequestInfo               info                 = 3;
-    RequestInitChain          init_chain           = 4;
-    RequestQuery              query                = 5;
-    RequestBeginBlock         begin_block          = 6;
-    RequestCheckTx            check_tx             = 7;
-    RequestDeliverTx          deliver_tx           = 8;
-    RequestEndBlock           end_block            = 9;
-    RequestCommit             commit               = 10;
-    RequestListSnapshots      list_snapshots       = 11;
-    RequestOfferSnapshot      offer_snapshot       = 12;
-    RequestLoadSnapshotChunk  load_snapshot_chunk  = 13;
-    RequestApplySnapshotChunk apply_snapshot_chunk = 14;
+    RequestEcho                echo                  = 1;
+    RequestFlush               flush                 = 2;
+    RequestInfo                info                  = 3;
+    RequestInitChain           init_chain            = 4;
+    RequestQuery               query                 = 5;
+    RequestBeginBlock          begin_block           = 6 [deprecated = true];
+    RequestCheckTx             check_tx              = 7;
+    RequestDeliverTx           deliver_tx            = 8 [deprecated = true];
+    RequestEndBlock            end_block             = 9 [deprecated = true];
+    RequestCommit              commit                = 10;
+    RequestListSnapshots       list_snapshots        = 11;
+    RequestOfferSnapshot       offer_snapshot        = 12;
+    RequestLoadSnapshotChunk   load_snapshot_chunk   = 13;
+    RequestApplySnapshotChunk  apply_snapshot_chunk  = 14;
+    RequestPrepareProposal     prepare_proposal      = 15;
```

Comment:
Tag 15 was removed in 82e4693c, we should probably not re-use it.

Was:

    RequestApplySnapshotChunk apply_snapshot_chunk = 15;

We can use a [reserved clause](https://developers.google.com/protocol-buffers/docs/proto3#reserved) to avoid old tags in the future. 

I'm not sure if this is the only one, but it's the only one I found offhand.

Reply:
changing the number format in major releases, while not preferable, should be fine. Apps should refetch all the proto files when updating otherwise they could encounter other errors.

Reply:
Yeah it seems like we actually removed `set_option`:
```
RequestSetOption          set_option           = 4;
```
and pushed everything up so really we should reserve 4 but I think we can get away with this (we already did this in 0.34).